### PR TITLE
Updated training page to allow the user to label the image

### LINKED ISSUE #97: Sketch out the train page to allow for upload of a file and categorize an individual
**User statement**
*As a user, I would like to specify which individual a certain reference image belongs to.*

**Proposed solution**
Implement some dropdown/multi-select along with a text field with bootstrap to allow the user to indicate a specific individual.

**Development notes**
Add any other notes, context, or screenshots relevant to the ticket here

### PR DESCRIPTION
**Pull request by rudy-patel**
### What is it?
This PR updates the Train page to allow the user to select and label the reference images they are uploading.

Closes #97.

### Screenshots
image

### CODE REVIEW

File: server/Forms.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from flask_wtf import FlaskForm
-from wtforms import StringField, PasswordField, BooleanField, EmailField
```
Comment (rudy-patel) [inline server/Forms.py:2]: Removed this unused import.

File: server/main.py [hunk 2]
```diff
@@ -178,23 +178,21 @@ def cameras():
...
+    if request.method == "POST":
+        file = request.files['file']
+        if file.filename != '':
+            if file:
+                file.filename = secure_filename(file.filename)
+                output = send_to_s3(file, "lfiasimagestore")
+                return str(output)
+        else:
+            return redirect("myapp.home")
+
```
Comment (rudy-patel) [inline server/main.py:193]: I combined the two separate `/train` API routes and consolidated into just one function.

File: server/templates/events.html [hunk 3]
```diff
@@ -7,7 +7,7 @@ <h3>Events</h3>
         <table class="table table-bordered table-striped events-table" style="width: 100%">
             <thead>
               <tr>
-                <th>Sensor</th>
+                <th>Camera</th>
```
Comment (rudy-patel) [inline server/templates/events.html:10]: I also just went ahead and changed the column headings on the events page from Sensor to Camera. We aren't using any sensors so this makes more sense.

File: server/Forms.py [hunk 4]
```diff
@@ -19,4 +20,10 @@ class SignUpForm(FlaskForm):
...
                             validators=[InputRequired(), Length(min=6, max=80)],
-                            render_kw={"placeholder": "Password"})
\ No newline at end of file
+                            render_kw={"placeholder": "Password"})
+
+class TrainingForm(FlaskForm):
+    file = FileField('image', validators=[
+        FileRequired(),
+        FileAllowed(['jpg', 'jpeg'], 'Allowed file formats: .jpg, .jpeg!')
+    ])
```
Comment (rudy-patel) [inline server/Forms.py:29]: This is the file field in the form, you can easily add validation here using the built-in functions.

**Review by kaden-dreger — DISMISSED** [review]
Looks good

File: server/Forms.py [hunk 5]
```diff
@@ -19,4 +20,32 @@ class SignUpForm(FlaskForm):
...
+                        render_kw={"placeholder": "New person name"})
+
+    def validate(self, extra_validators=None):
+        if super().validate(extra_validators):
+
+            if not ((self.personSelect.data and not self.newPersonName.data) or (not self.personSelect.data and self.newPersonName.data)):
+                self.newPersonName.errors.append('You must choose an existing person OR provide a new name, not both!')
+                return False
+            else:
+                return True
```
Comment (rudy-patel) [inline server/Forms.py:49]: This is a custom validation function. In this form, the user should **either** select an existing user OR enter a new name, but not **both**.

File: server/templates/train.html [hunk 6]
```diff
@@ -1,7 +1,30 @@
...
+  <div class="row justify-content-center">
+    <h3>Upload a New Reference Image</h3>
+  </div>
+
+  <form class="form-uploadImage" method="POST" action ="/train" enctype="multipart/form-data">
+    <div class="form-group">
+      {{ form.hidden_tag() }}
+      {{ wtf.form_field(form.file) }}
+      {{ wtf.form_field(form.personSelect) }}
+      {{ wtf.form_field(form.newPersonName) }}
```
Comment (rudy-patel) [inline server/templates/train.html:16]: The new form style on the Train page

File: server/templates/train.html [hunk 7]
```diff
@@ -1,7 +1,30 @@
...
+      <p><button type="submit" class="btn btn-primary">Upload Training Data</button></p>
+    </div>
+  </form>
+
+  {% for message in get_flashed_messages() %}
+        <div class="alert alert-warning">
+            <button type="button" class="close" data-dismiss="alert">&times;</button>
+            {{ message }}
+        </div>
+  {% endfor %}
```
Comment (rudy-patel) [inline server/templates/train.html:26]: This block flashes any error/success messages on the form.

**Review by kaden-dreger — APPROVED** [review]

**Pull request merged**